sdk/daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/UpgradesSpec.scala

```diff
@@ -756,6 +767,19 @@ abstract class UpgradesSpec(val suffix: String)
     }
   }
 
+  def testPackagePair(
```

Comment:
Can we avoid this duplication via some sort of type parameter that has trait "ReadToPackageId", with an instance for String and for (PackageId, ByteString)?

Reply:
I am not sure about which duplication you are speaking about,
The second `testPackagePair` calls the first one, so there is really nothing to factorize.